osprey_flask_app/routes.py

```diff
@@ -44,17 +46,64 @@ def osprey_route():
     """
     args = request.args
     arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
     try:
-        outpath_url = requests.get(outpath)
-    except requests.exceptions.ConnectionError as e:
-        raise
-
-    with NamedTemporaryFile(suffix=".nc", dir="/tmp") as outfile:
-        outfile.write(outpath_url.content)
-        return send_file(
-            outfile.name,
-            mimetype="application/x-netcdf",
-            as_attachment=True,
-            attachment_filename=os.path.basename(outpath),
+        inputs_are_valid(arg_dict)
+    except Exception as e:
+        return Response(str(e), status=400)
+
+    rvic_thread = pool.submit(run_full_rvic, arg_dict)
+    job_threads.append(rvic_thread)
+
+    job_id = str(uuid.uuid4())  # Generate unique id for tracking request
+    job_ids.append(job_id)
+    return Response(
+        "RVIC Process started. Check status: "
+        + url_for("osprey.status_route", job_id=job_id),
+        status=202,
+    )
+
+
+@osprey.route("/status/<job_id>", methods=["GET"])
+def status_route(job_id):
+    """Provide route to check status of RVIC process."""
+    if job_id not in job_ids:
+        return Response("Process with this id does not exist.", status=201)
+
+    job_index = job_ids.index(job_id)
+    job_thread = job_threads[job_index]
+    if not job_thread.done():
+        return Response("Process is still running.", status=201)
+    else:
+        return Response(
+            "Process completed. Get output: "
+            + url_for("osprey.output_route", job_id=job_id),
+            status=200,
         )
+
+
+@osprey.route("/output/<job_id>", methods=["GET"])
+def output_route(job_id):
+    """Provide route to get streamflow output of RVIC process."""
+    if job_id not in job_ids:
+        return Response("Process with this id does not exist.", status=404)
+
+    job_index = job_ids.index(job_id)
+    job_thread = job_threads[job_index]
+
+    # Remove id from list of executing jobs
+    job_ids.remove(job_id)
+    job_threads.remove(job_thread)
+
+    job_exception = job_thread.exception()
+    if job_exception is not None:
+        return Response("Process has failed. " + job_exception, status=404)
+
+    try:
+        outpath = job_thread.result()
+        outpath_response = requests.get(outpath)
+    except requests.exceptions.ConnectionError as e:
+        return Response("Process has failed. " + e, status=404)
+
+    return Response(
+        f"Process successfully completed. Output url: {outpath}", status=302
```

Comment:
Maybe we could return the response as a JSON object, so that users can programmatically access the output URL? Something like:

```python3
import json
...
return Response(
   json.dumps({
        message: "Process successfully completed"
        url: outpath
    }),
    content_type="application/json",
    status=302
)
```

Or better yet, just set the "Location" header, which browsers will redirect to.

```python3
return Response(... headers={"Location": outpath})
```